level1c4pps/eumgacfdr2pps_lib.py

```diff
@@ -195,7 +196,31 @@ def set_header_and_band_attrs(scene):
     return nimg
 
 
-def process_one_file(eumgacfdr_file, out_path='.', reader_kwargs=None, engine='h5netcdf'):
+def crop(scene, start_line, end_line, time_key='scanline_timestamps'):
+    """Crop datasets and update start_time end_time objects."""
+    start_time_dt64 = scene[time_key].values[start_line]
+    end_time_dt64 = scene[time_key].values[end_line-1]
+    start_time = dt64_to_datetime(start_time_dt64)
+    end_time = dt64_to_datetime(end_time_dt64)
+    print("time to crop the data")
+    for ds in scene.keys():
+        print(ds)
+        if 'y' in scene[ds].dims:
+            scene[ds] = scene[ds].isel(y=slice(start_line, end_line))
```

Comment:
I think it needs to be `end_line + 1` for the end line to be included. My fault initially, sorry.

Reply:
And the corresponding end timestamp would be `end_time_dt64 = scene[time_key].values[end_line]`